tests/test_config_generation.py

```diff
@@ -0,0 +1,51 @@
+import os
+import pathlib
+import unittest
+from unittest.mock import patch
+
+import yaml
+
+import data_pipelines_cli.config_generation as cgen
+from data_pipelines_cli.cli_constants import profiles_build_path
+
+
+class TestConfigGeneration(unittest.TestCase):
+    def setUp(self) -> None:
+        self.maxDiff = None
+        self.current_dir_path = pathlib.Path(__file__).parent
+        self.profiles_path = None
+
+    def tearDown(self) -> None:
+        if self.profiles_path:
+            os.remove(self.profiles_path)
+            os.rmdir(self.profiles_path.parent)
+            os.rmdir(self.profiles_path.parent.parent)
+
+    def _test_generation(self, env: str, profile_type: str):
+        with patch(
+            "data_pipelines_cli.cli_constants.BUILD_DIR", self.current_dir_path
+        ), patch(
+            "data_pipelines_cli.config_generation.BUILD_DIR",
+            self.current_dir_path,
+        ):
+            cgen.generate_profiles_yml(env)
+
+            self.profiles_path = profiles_build_path(env)
+            with open(self.profiles_path, "r") as generated, open(
+                self.current_dir_path.joinpath(
+                    "example_profiles", f"{env}_{profile_type}.yml"
+                ),
+                "r",
+            ) as prepared:
+                self.assertDictEqual(
+                    yaml.safe_load(generated), yaml.safe_load(prepared)
+                )
+
+    def test_dev_bigquery_generation(self):
```

Comment:
nb: maybe subtests? https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

Reply:
changed to subtests